src/frequenz/channels/_latest_value_cache.py

```diff
@@ -48,15 +57,56 @@ def __str__(self) -> str:
         return "<no value received yet>"
 
 
-class LatestValueCache(typing.Generic[T_co]):
+class LatestValueCache(typing.Generic[T_co, HashableT]):
```

Comment:
Maybe not a biggie, but this introduces a new requirements for values to cache, that will be required even if `key` is not desired or used. I guess in most cases this will not be an issue, and I guess there is no solution unless we have a separate non-keyed `LatestValueCache`, so it is probably fine.

Reply:
If no key is specified, this will be `None`.  Apparantly `None` is hashable.